### ISSUE TITLE
Character Roster not showing favorite or recent characters

### ISSUE THREAD

**Issue by Nobnobb** [chummer5a/chummer5a#2454]
### Environment
Chummer Version: 5.197.25
Environment: Windows 10 64-Bit
Runtime: .NET 4.5.1

### Expected behaviour
Recently opened characters show up in the character roster under the proper list, from which one can mark it as favorite to always have it listed. Double-click on the character opens it up.

### Actual behaviour
The favorite character list disappeared entirely, even though I did not replace, rename, move or delete any characters that were listed there. After opening a character up manually, it does not end up in the Recent-List and if I try to favorite any save-file, it disappears again after restarting Chummer. As such, the Character Roster is always entirely empty when I open it.

### Steps to reproduce the behaviour
Seems to happen at all times, no matter which character or location is used.

### Link to affected character file
No characters.

**Comment by DelnarErsike** [comment]
The way the character roster code works, if a character has been added to a list, it will only *not* show the character if it either cannot find the character file, it cannot access the character file, or it encounters an error while reading the character file.

Could you check the contents of the registry key `HKEY_CURRENT_USER\Software\Chummer5\Mru` to see if the characters were actually added to your recent/favorites lists? If they were, we can narrow down the issue to either line 920 of `clsOptions.cs` or lines 216 or 220 of `frmCharacterRoster.cs`.

**Comment by Nobnobb** [comment]
There are several mru1-4 entries present, all leading to the correct path of the characters I recently used, as well as one stickymru1 entry, which I assume could be the favorite entry. So it looks good on that end.

**Comment by DelnarErsike** [comment]
Hmm, well, the MRUs exist even after things are changed, so the problem has to be at lines 216 or 220 of `frmCharacterRoster.cs`. I've added code to the character roster in the latest commit to help debug the issue. Let me know how the roster works in the next nightly (or with the current dev build if you compile your own builds).

**Comment by Nobnobb** [comment]
Now that's interesting. The roster itself shows characters again, but I cannot open from them. Instead, the character is displayed in red with an (error) behind it and when hovering over it, he's giving me this: 
![chummer5_2018-02-08_02-12-04](https://user-images.githubusercontent.com/35600529/35950234-d9d2eb88-0c75-11e8-98a4-1acd23667ab4.png)

**Comment by DelnarErsike** [comment]
Hmm, something got messed up with the savefile's character encoding, or Chummer5 can't properly recognize that everything should be in UTF-8. If you attempt to load an error'ed character from the dropdown list of most recently used characters instead of from the character roster, do they load correctly?

**Comment by Nobnobb** [comment]
Yes, same if I try to open them manually. 
Also, it does not happen for all characters either. I tried a total of six characters, four of them error out, two work fine, even over the roster. I cannot see any certain property the working ones all share, it's a wild mix of races and details where almost nothing is a like.

They all are perfectly fine in itself and loadable using open or the recent character list outside of the roster, without any bugs or weird things happening when I opened them up.
